luarules/gadgets/cmd_build_bugger_off.lua

```diff
@@ -0,0 +1,208 @@
+				local searchRadius		= cachedUnitDefs[builtUnitDefID].radius + SEARCH_RADIUS_OFFSET
+				local interferingUnits	= Spring.GetUnitsInCylinder(targetX, targetZ, searchRadius)
+
+				-- Escalate the radius every update. We want to send units away the minimum distance, but  
+				-- if there are many units in the way, they may cause a traffic jam and need to clear more room.
+				builderRadiusOffsets[builderID] = builderRadiusOffsets[builderID] + BUGGEROFF_RADIUS_ESCALATION_INCREMENT
+
+				for _, interferingUnitID in ipairs(interferingUnits) do
+					if builderID ~= interferingUnitID and visited[interferingUnitID] == nil then
+						-- Only buggeroff from one build site at a time
+						visited[interferingUnitID] = true
+						local unitX, _, unitZ = Spring.GetUnitPosition(interferingUnitID)
+						if shouldIssueBuggeroff(builderTeam(builderID), interferingUnitID, targetX, targetY, targetZ, buggerOffRadius) then
+							local sendX, sendZ = math.closestPointOnCircle(targetX, targetZ, buggerOffRadius, unitX, unitZ)
+
+							if Spring.TestMoveOrder(Spring.GetUnitDefID(interferingUnitID), sendX, targetY, sendZ) then 
+								Spring.GiveOrderToUnit(interferingUnitID, CMD.INSERT, {0, CMD.MOVE, CMD.OPT_INTERNAL, sendX, targetY, sendZ}, CMD.OPT_ALT )
+							end
+						end
+					end
+				end
+
+			elseif isBuilding then
+				-- We want to keep updating in case he's got another job nearby
+				builderRadiusOffsets[builderID] = 0
+			end
+		end
+
+		if frame % SLOW_UPDATE_FREQUENCY ~= 0 then
+			return
+		end
+
+		for builderID, _ in pairs(slowUpdateBuilders) do
+			local builderCommands   = Spring.GetUnitCommands(builderID, -1)
+			local hasBuildCommand, buildCommandFirst = false, false
+			local targetX, targetZ  = 0, 0
+
+			if builderCommands ~= nil then
+				for idx, command in pairs(builderCommands) do
+					if command.id < 0 then
+						hasBuildCommand = true
+						if idx == 1 then
+							buildCommandFirst = true
+							targetX, targetZ  = command.params[1], command.params[3]
+						end
+					end
+				end
+			end
+
+			local isBuilding  = false
+			if Spring.GetUnitIsBuilding(builderID) then isBuilding = true end
+
+			local x, _, z = Spring.GetUnitPosition(builderID)
+			if hasBuildCommand == false then
+				print("Clearing watched builder slow" .. frame)
+				slowUpdateBuilders[builderID]   = nil
+				builderRadiusOffsets[builderID] = nil
+			elseif buildCommandFirst and isBuilding == false and math.distance2d(targetX, targetZ, x, z) <= FAST_UPDATE_RADIUS then
+				print("Promote to fast" .. frame)
+				slowUpdateBuilders[builderID]   = nil
+				watchedBuilders[builderID]		= true
+			end
+		end
+	end
+
+	function gadget:MetaUnitAdded(unitID, unitDefID, unitTeam)
+		if cachedUnitDefs[unitDefID].isBuilder then
+			cachedBuilderTeams[unitID] = unitTeam
+		end
+	end
+
+	function gadget:MetaUnitRemoved(unitID, unitDefID, unitTeam)
+		if cachedUnitDefs[unitDefID].isBuilder then
+			cachedBuilderTeams[unitID] = nil
+		end
```

Comment:
The check doesn't do anything since other units are nil, and if they somehow weren't nil it would've been a good idea to nil them either way.
```suggestion
		cachedBuilderTeams[unitID] = nil
```